### PR TITLE
spdlog

### PR DESCRIPTION
**Pull request by endJunction**
Replacing logog logger with modern and powerfull spdlog logger (backed up by the fmt formatting library!).
The most obvious advantage is the type safety (coming from the fmt library). (Indeed I've found two or three places, where the garbage was output.)

1. [ ] Feature description was added to the [changelog](https://github.com/ufz/ogs/wiki/Release-notes-6.3.1)
2. [x] Tests covering your feature were added? (`ogs` :grin: )
3. [ ] Any new feature or behavior change was documented?

- [ ] Linking, not header only.

### CODE REVIEW

**Comment by endJunction** [comment]
I'll do some cleanup of the commits, of course.
Before we go into detailed discussions, I'd like that you try it out and tell here what you think about the output (which of course has slightly changed).

**Comment by renchao-lu** [comment]
I prefer more to screen out the full message instead of abbreviation.
I -> INFO:
W -> WARN:
E -> ERROR:

Screenshot from 2020-03-31 09-27-51

This is solely my taste. For you reference.

**Comment by TomFischer** [comment]
@endJunction Thanks for the big effort!

The changes passed the test on JUWELS. The new log files are slightly bigger (943851 bytes (spdlog) vs. 900077 bytes (master)) in my test example. My log files evaluation scripts worked without changes.

For discussion: IMHO I would  omit the 'I  ' for all output that doesn't fit in the other categories (DEBUG, WARN, and ERROR). And, as @renchao-lu already suggested, I would also prefer more readability (at least for the users) prefixes.
 - I -> 
 - D -> DEB
 - W -> WARN
 - E -> ERR

Or, to make it clearer that this part of the message is an abbreviation:
 - I -> 
- D -> [D]
 - W -> [W]
 - E -> [E]

**Comment by endJunction** [comment]
OK. I have cleaned up the commits. First selective reviews were done (thanks @renchao-lu, @TomFischer, and @bilke).
Have fun!

**Comment by bilke** [comment]
⏩ when you have decided on a linking pattern.

**Comment by endJunction** [comment]
Linkage is still a problem, but needed for shorter build times. Working on it.

File: MeshGeoToolsLib/AppendLinesAlongPolyline.cpp [hunk 1]
```diff
@@ -52,7 +52,7 @@ std::unique_ptr<MeshLib::Mesh> appendLinesAlongPolylines(
         if (vec_nodes_on_ply.empty()) {
             std::string ply_name;
             ply_vec.getNameOfElementByID(k, ply_name);
-            INFO("No nodes found on polyline %s", ply_name.c_str());
+            INFO("No nodes found on polyline {:s}", ply_name.c_str());
```
Comment (TomFischer) [inline MeshGeoToolsLib/AppendLinesAlongPolyline.cpp:55]: Maybe a stupid question, bu is `.c_str()` still needed?

File: MeshGeoToolsLib/BoundaryElementsAtPoint.cpp [hunk 2]
```diff
@@ -52,9 +52,11 @@ BoundaryElementsAtPoint::BoundaryElementsAtPoint(
...
     {
         OGS_FATAL(
-            "BoundaryElementsAtPoint: the mesh node searcher found %d points "
-            "near the requested point (%f, %f, %f) in the mesh, while exactly "
-            "one is expected. Node  (id=%d) (%f, %f, %f) has distance %f.",
+            "BoundaryElementsAtPoint: the mesh node searcher found {:d} points "
+            "near the requested point ({:f}, {:f}, {:f}) in the mesh, while "
+            "exactly "
+            "one is expected. Node  (id={:d}) ({:f}, {:f}, {:f}) has distance "
+            "{:f}.",
```
Comment (TomFischer) [inline MeshGeoToolsLib/BoundaryElementsAtPoint.cpp:59]: Changes could be better formatted.

File: MeshLib/IO/MPI_IO/NodePartitionedMeshReader.cpp [hunk 3]
```diff
@@ -217,9 +219,11 @@ void NodePartitionedMeshReader::readPropertiesBinary(
...
     if (!is)
     {
-        WARN("Could not open file '%s'.\n"
-             "\tYou can ignore this warning if the mesh does not contain %s-"
-             "wise property data.", fname_cfg.c_str(), item_type.data());
+        WARN(
+            "Could not open file '{:s}'.\n"
+            "\tYou can ignore this warning if the mesh does not contain {:s}-"
+            "wise property data.",
+            fname_cfg.c_str(), item_type.data());
```
Comment (TomFischer) [inline MeshLib/IO/MPI_IO/NodePartitionedMeshReader.cpp:226]: `item_type` is already a `string`. Maybe, calling the method `data()` isn't necessary?

File: MeshLib/IO/MPI_IO/NodePartitionedMeshReader.cpp [hunk 4]
```diff
@@ -272,9 +277,10 @@ void NodePartitionedMeshReader::readPropertiesBinary(
     is.open(fname_val.c_str(), std::ios::binary | std::ios::in);
     if (!is)
     {
-        ERR("Could not open file '%s'\n."
-            "\tYou can ignore this warning if the mesh does not contain %s-"
-             "wise property data.", fname_val.c_str(), item_type.data());
+        ERR("Could not open file '{:s}'\n."
+            "\tYou can ignore this warning if the mesh does not contain {:s}-"
+            "wise property data.",
+            fname_val.c_str(), item_type.data());
```
Comment (TomFischer) [inline MeshLib/IO/MPI_IO/NodePartitionedMeshReader.cpp:283]: `item_type` is already a `string`. Maybe, calling the method `data()` isn't necessary?

File: MeshLib/MeshSearch/NodeSearch.cpp [hunk 5]
```diff
@@ -13,7 +13,7 @@
 #include <memory>
 #include <set>
 
-#include <logog/include/logog.hpp>
+#include "BaseLib/Logging.h"
```
Comment (TomFischer) [inline MeshLib/MeshSearch/NodeSearch.cpp:16]: As far as I can see there isn't any logging output in this file - maybe the include can be removed.

File: MeshLib/MeshSearch/ElementSearch.cpp [hunk 6]
```diff
@@ -10,7 +10,7 @@
 
 #include "ElementSearch.h"
 
-#include <logog/include/logog.hpp>
+#include "BaseLib/Logging.h"
```
Comment (TomFischer) [inline MeshLib/MeshSearch/ElementSearch.cpp:13]: As far as I can see there isn't any logging output in this file - maybe the include can be removed.

File: MeshGeoToolsLib/AppendLinesAlongPolyline.cpp (hunk 1 above)
Comment (endJunction) [reply MeshGeoToolsLib/AppendLinesAlongPolyline.cpp:55]: No. Could be removed, but then everywhere.

File: MeshGeoToolsLib/BoundaryElementsAtPoint.cpp (hunk 2 above)
Comment (endJunction) [reply MeshGeoToolsLib/BoundaryElementsAtPoint.cpp:59]: Manual work, unfortunately. Sometimes the breaks are for better readability of the strings, and should be kept as they are. So no automatic formatting possible.

File: MeshLib/IO/MPI_IO/NodePartitionedMeshReader.cpp (hunk 3 above)
Comment (endJunction) [reply MeshLib/IO/MPI_IO/NodePartitionedMeshReader.cpp:226]: Same as c_str

File: MeshLib/MeshSearch/NodeSearch.cpp (hunk 5 above)
Comment (endJunction) [reply MeshLib/MeshSearch/NodeSearch.cpp:16]: Yeah. This is another PR for header includes cleanup.

**Pull request merged**